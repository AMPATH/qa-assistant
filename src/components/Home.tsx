import { useEffect, useState } from 'react'
import { Link } from 'react-router-dom'
import Header from './Header'

// const api_url = 'openmrs/ws/rest/v1/patient?q=Sarah&v=default&limit=1'

interface Result {
    [key: string]: any
}
const Home = () => {
    const [searchParams, setSearchParams] = useState<string>('')
    const [results, setResults] = useState<Result[]>()


    const handleChange = (event: any) => {
        const value = event.target.value;
        setSearchParams(value)
    }

    const userInformation = localStorage.getItem("authenticated")
    if(userInformation !== "true"){
        window.location.href = "/login"
    }

    useEffect(() => {
        let timer: number;
        if(searchParams && searchParams.trim() && searchParams.length > 0) {
            timer = window.setTimeout(() => {
                fetch(`openmrs/ws/rest/v1/patient?q=${searchParams}&v=default&limit=1`, {
                    mode: 'no-cors',
                    method: "GET",
                    redirect: 'follow'
                }).then(res => res.json())
                .then(data => {
                    if(!data) {
                        return
                    } 
                    setResults(data)
                    setSearchParams('')
                })
            }, 2000)
        }
        return () => clearTimeout(timer)
    }, [searchParams])

    // if(results) {
    //     console.log(new Date(results?.results[0].person.birthdate).toLocaleDateString())
    //     const convertString = results?.results[0].display.split(' ').slice(2).join()
    //     // const splitString = convertString.split(' ').slice(2).join()
    //     console.log(convertString)
    // }

  return (
    <div>
       <Header />
        <div className='w-[80%] mx-auto pt-4 mb-10'>
            <div className='w-[80%] mx-auto m-4 flex gap-4'>
                <input value={searchParams} onChange={handleChange} className='bg-searchColor font-semi-bold outline-none w-full p-2 text-center text-xl border rounded-sm' type="text" placeholder='Search Patient'/>
                <div className='m-2'>
                <select className='p-4 outline-none'>
                    <option >Name</option>
                    <option>Patient Identifier</option>
                </select>
                </div>
            </div>
        </div>
        <div className='w-[80%] mx-auto mt-32'>
            <div></div>
            <div className='flex mx-auto w-[90%] justify-between rounded-md bg-slate-200 p-4'>
                <div>Name</div>
                <div>Age</div>
                <div>Gender</div>
                <div>Date of birth</div>
            </div>
            <div className='flex mx-auto w-[90%] justify-between p-4 cursor-pointer mt-1'>
                {results ? (
                    <>
                   <Link to={`${results?.results[0]?.person.uuid}`}>
                   <div className='hover:text-blue-600'>{results?.results[0]?.display.split(' ').slice(2).join(' ')}</div>
                   </Link>
                    <div>{results?.results[0]?.person.age}</div>
                    <div>{results?.results[0]?.person?.gender}</div>
                    <div>{results?.results[0]?.person.birthdate && new Date(results?.results[0]?.person.birthdate).toLocaleDateString()}</div>
                   </> 
                ) : (
                    <p>Search a patient!!</p>
                )}
            </div>
        </div>
    </div>
  )
}

export default Home
