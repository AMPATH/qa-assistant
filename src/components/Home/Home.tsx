import { useEffect } from "react";
import { Link, useNavigate } from "react-router-dom";
import Header from "../Header/Header";
import swal from "sweetalert";

const Home = () => {
  const menuItems = [];

  const cardStyle =
    "rounded-md overflow-hidden shadow-lg bg-white h-56 hover:cursor-pointer transition duration-300 ease-in-out hover:scale-110";
  const textStyle = "text-center font-bold text-xl mt-6";

  return (
    <div className="md:h-screen">
      <Header />
      <div className="flex items-center justify-center bg-gray-50 lg:h-5/6 pt-3">
        <div className="flex flex-col border-2 w-3/6 md:h-full lg:h-5/6 rounded-md shadow-lg bg-gray-200">
          <div className="text-4xl text-center mt-6 font-bold text-blue-500">
            <h1>QA Assistant</h1>
          </div>
          <div className="grid grid-cols-1 gap-6 md:grid-cols-2 lg:grid-cols-3 mt-12 h-3/4 p-10 mt-2">
            <div className={cardStyle}>
              <Link to={"/patient-search"}>
                <img
                  className="w-full"
                  src="src/public/search.png"
                  alt="patient search"
                ></img>
                <h1 className={textStyle}>Patient Search</h1>
              </Link>
            </div>
            <div className={cardStyle}>
              <img
                className="w-2/4 h-2/4 sm:ml-16 sm:mr-8 md:ml-8 md:ml-8 lg:ml-16 lg:mr-20 mt-8"
                src="src/public/graph.jpeg"
                alt="data dump"
              ></img>
              <h1 className={textStyle}>Data Dump Upload</h1>
            </div>
            <div className={cardStyle}></div>
            <div className={cardStyle}></div>
            <div className={cardStyle}></div>
            <div className={cardStyle}></div>
          </div>
        </div>
      </div>
    </div>
  );
};

export default Home;
