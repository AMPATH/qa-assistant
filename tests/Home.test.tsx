import React from 'react'
import { render } from '@testing-library/react'
import Home from '../src/components/Home'
import { MemoryRouter as Router } from 'react-router-dom';
// import fetch from "node-fetch";
import { describe, it, assertType, expectTypeOf} from 'vitest'


describe('Home component', () => {
    const mockPatient = [
        {
            uuid: '1232332asasasa',
            identifier: '123333',
            patient_name: 'john doe',
            gender: 'M',
            age: 23
        }
    ]

    const handleAdvancedFiltering = (): any => {
        return mockPatient
    }

    const setCurrentPage = (number: number) => {
        return number
    }

    const paginate = (pageNumber: number) => setCurrentPage(pageNumber)

    it('renders the Home component', () => {
        render(
            <Router>
                <Home />
            </Router>
        )
    })

    it('renders home snaptshot correctly', () => {
        const result = render(<Router>
            <Home />
        </Router>)
      expect(result).toMatchSnapshot()
    })

    it('expects handle filtering to be executed', () => {
        expectTypeOf(handleAdvancedFiltering).toBeFunction()
        expectTypeOf(handleAdvancedFiltering).parameter(0).toMatchTypeOf()
    })

    it('expects a paginate function', () => {
        expectTypeOf(paginate).toBeFunction()
        expectTypeOf(paginate).parameter(0).toMatchTypeOf<number>()
    })
});