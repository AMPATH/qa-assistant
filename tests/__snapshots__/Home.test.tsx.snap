// Vitest Snapshot v1

exports[`Home component > renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <header
          class="bg-white shadow-md flex justify-between items-center"
        >
          <span
            class="cursor-pointer ml-3"
          >
            <a
              href="/"
            >
              <img
                src="/public/logo/ampath_Logo.png"
                width="180"
              />
            </a>
          </span>
          <nav
            class="md:mr-12"
          >
            <ul
              class="md:flex items-center gap-20"
            >
              <li
                class="text-sm"
              >
                Logged in as 
                <span
                  class="uppercase"
                >
                  <strong />
                </span>
              </li>
              <li
                class="text-lg flex gap-2 items-center hover:shadow-lg hover:bg-slate-300 p-2 rounded-sm cursor-pointer"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                  <path
                    d="M17 7l-1.41 1.41L18.17 11H8v2h10.17l-2.58 2.58L17 17l5-5zM4 5h8V3H4c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h8v-2H4V5z"
                  />
                </svg>
                Log out
              </li>
            </ul>
          </nav>
        </header>
      </div>
      <div
        class="bg-themeColor overflow-y-auto h-screen pt-10"
      >
        <div
          class="w-[80%] ml-[12%]"
        >
          <div
            class="w-[90%] mx-auto"
          >
            <div
              class="md:flex gap-10 m-4 mx-auto w-[95%] md:ml-20"
            >
              <input
                class="py-2 px-4 md:w-[60%] w-full outline-none rounded-xl shadow-lg"
                placeholder="Search patient by name"
                type="text"
                value=""
              />
              <div
                class="flex md:gap-11 gap-3 mt-4"
              >
                <button
                  class="bg-blue-800/70 text-lg text-white py-2 px-12 rounded-xl hover:bg-white border hover:border-blue-800/70 hover:text-blue-800/70"
                >
                  Search
                </button>
                <button
                  class="bg-slate-50 text-lg text-red-600 border hover:border-red-500 hover:font-bold border-gray-300 py-2 px-12 rounded-xl"
                >
                  Reset
                </button>
              </div>
            </div>
          </div>
          <p
            class="text-lg ml-8"
          >
            Search for a patient
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <header
        class="bg-white shadow-md flex justify-between items-center"
      >
        <span
          class="cursor-pointer ml-3"
        >
          <a
            href="/"
          >
            <img
              src="/public/logo/ampath_Logo.png"
              width="180"
            />
          </a>
        </span>
        <nav
          class="md:mr-12"
        >
          <ul
            class="md:flex items-center gap-20"
          >
            <li
              class="text-sm"
            >
              Logged in as 
              <span
                class="uppercase"
              >
                <strong />
              </span>
            </li>
            <li
              class="text-lg flex gap-2 items-center hover:shadow-lg hover:bg-slate-300 p-2 rounded-sm cursor-pointer"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M17 7l-1.41 1.41L18.17 11H8v2h10.17l-2.58 2.58L17 17l5-5zM4 5h8V3H4c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h8v-2H4V5z"
                />
              </svg>
              Log out
            </li>
          </ul>
        </nav>
      </header>
    </div>
    <div
      class="bg-themeColor overflow-y-auto h-screen pt-10"
    >
      <div
        class="w-[80%] ml-[12%]"
      >
        <div
          class="w-[90%] mx-auto"
        >
          <div
            class="md:flex gap-10 m-4 mx-auto w-[95%] md:ml-20"
          >
            <input
              class="py-2 px-4 md:w-[60%] w-full outline-none rounded-xl shadow-lg"
              placeholder="Search patient by name"
              type="text"
              value=""
            />
            <div
              class="flex md:gap-11 gap-3 mt-4"
            >
              <button
                class="bg-blue-800/70 text-lg text-white py-2 px-12 rounded-xl hover:bg-white border hover:border-blue-800/70 hover:text-blue-800/70"
              >
                Search
              </button>
              <button
                class="bg-slate-50 text-lg text-red-600 border hover:border-red-500 hover:font-bold border-gray-300 py-2 px-12 rounded-xl"
              >
                Reset
              </button>
            </div>
          </div>
        </div>
        <p
          class="text-lg ml-8"
        >
          Search for a patient
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Home component > renders home cocorrectly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <header
          class="bg-white shadow-md flex justify-between items-center"
        >
          <span
            class="cursor-pointer ml-3"
          >
            <a
              href="/"
            >
              <img
                src="/public/logo/ampath_Logo.png"
                width="180"
              />
            </a>
          </span>
          <nav
            class="md:mr-12"
          >
            <ul
              class="md:flex items-center gap-20"
            >
              <li
                class="text-sm"
              >
                Logged in as 
                <span
                  class="uppercase"
                >
                  <strong />
                </span>
              </li>
              <li
                class="text-lg flex gap-2 items-center hover:shadow-lg hover:bg-slate-300 p-2 rounded-sm cursor-pointer"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                  <path
                    d="M17 7l-1.41 1.41L18.17 11H8v2h10.17l-2.58 2.58L17 17l5-5zM4 5h8V3H4c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h8v-2H4V5z"
                  />
                </svg>
                Log out
              </li>
            </ul>
          </nav>
        </header>
      </div>
      <div
        class="bg-themeColor overflow-y-auto h-screen pt-10"
      >
        <div
          class="w-[80%] ml-[12%]"
        >
          <div
            class="w-[90%] mx-auto"
          >
            <div
              class="md:flex gap-10 m-4 mx-auto w-[95%] md:ml-20"
            >
              <input
                class="py-2 px-4 md:w-[60%] w-full outline-none rounded-xl shadow-lg"
                placeholder="Search patient by name"
                type="text"
                value=""
              />
              <div
                class="flex md:gap-11 gap-3 mt-4"
              >
                <button
                  class="bg-blue-800/70 text-lg text-white py-2 px-12 rounded-xl hover:bg-white border hover:border-blue-800/70 hover:text-blue-800/70"
                >
                  Search
                </button>
                <button
                  class="bg-slate-50 text-lg text-red-600 border hover:border-red-500 hover:font-bold border-gray-300 py-2 px-12 rounded-xl"
                >
                  Reset
                </button>
              </div>
            </div>
          </div>
          <p
            class="text-lg ml-8"
          >
            Search for a patient
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <header
        class="bg-white shadow-md flex justify-between items-center"
      >
        <span
          class="cursor-pointer ml-3"
        >
          <a
            href="/"
          >
            <img
              src="/public/logo/ampath_Logo.png"
              width="180"
            />
          </a>
        </span>
        <nav
          class="md:mr-12"
        >
          <ul
            class="md:flex items-center gap-20"
          >
            <li
              class="text-sm"
            >
              Logged in as 
              <span
                class="uppercase"
              >
                <strong />
              </span>
            </li>
            <li
              class="text-lg flex gap-2 items-center hover:shadow-lg hover:bg-slate-300 p-2 rounded-sm cursor-pointer"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M17 7l-1.41 1.41L18.17 11H8v2h10.17l-2.58 2.58L17 17l5-5zM4 5h8V3H4c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h8v-2H4V5z"
                />
              </svg>
              Log out
            </li>
          </ul>
        </nav>
      </header>
    </div>
    <div
      class="bg-themeColor overflow-y-auto h-screen pt-10"
    >
      <div
        class="w-[80%] ml-[12%]"
      >
        <div
          class="w-[90%] mx-auto"
        >
          <div
            class="md:flex gap-10 m-4 mx-auto w-[95%] md:ml-20"
          >
            <input
              class="py-2 px-4 md:w-[60%] w-full outline-none rounded-xl shadow-lg"
              placeholder="Search patient by name"
              type="text"
              value=""
            />
            <div
              class="flex md:gap-11 gap-3 mt-4"
            >
              <button
                class="bg-blue-800/70 text-lg text-white py-2 px-12 rounded-xl hover:bg-white border hover:border-blue-800/70 hover:text-blue-800/70"
              >
                Search
              </button>
              <button
                class="bg-slate-50 text-lg text-red-600 border hover:border-red-500 hover:font-bold border-gray-300 py-2 px-12 rounded-xl"
              >
                Reset
              </button>
            </div>
          </div>
        </div>
        <p
          class="text-lg ml-8"
        >
          Search for a patient
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Home component > renders home snaptshot correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <header
          class="bg-white shadow-md flex justify-between items-center"
        >
          <span
            class="cursor-pointer ml-3"
          >
            <a
              href="/"
            >
              <img
                src="/public/logo/ampath_Logo.png"
                width="180"
              />
            </a>
          </span>
          <nav
            class="md:mr-12"
          >
            <ul
              class="md:flex items-center gap-20"
            >
              <li
                class="text-sm"
              >
                Logged in as 
                <span
                  class="uppercase"
                >
                  <strong />
                </span>
              </li>
              <li
                class="text-lg flex gap-2 items-center hover:shadow-lg hover:bg-slate-300 p-2 rounded-sm cursor-pointer"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                  <path
                    d="M17 7l-1.41 1.41L18.17 11H8v2h10.17l-2.58 2.58L17 17l5-5zM4 5h8V3H4c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h8v-2H4V5z"
                  />
                </svg>
                Log out
              </li>
            </ul>
          </nav>
        </header>
      </div>
      <div
        class="bg-themeColor overflow-y-auto h-screen pt-10"
      >
        <div
          class="w-[80%] ml-[12%]"
        >
          <div
            class="w-[90%] mx-auto"
          >
            <div
              class="md:flex gap-10 m-4 mx-auto w-[95%] md:ml-20"
            >
              <input
                class="py-2 px-4 md:w-[60%] w-full outline-none rounded-xl shadow-lg"
                placeholder="Search patient by name"
                type="text"
                value=""
              />
              <div
                class="flex md:gap-11 gap-3 mt-4"
              >
                <button
                  class="bg-blue-800/70 text-lg text-white py-2 px-12 rounded-xl hover:bg-white border hover:border-blue-800/70 hover:text-blue-800/70"
                >
                  Search
                </button>
                <button
                  class="bg-slate-50 text-lg text-red-600 border hover:border-red-500 hover:font-bold border-gray-300 py-2 px-12 rounded-xl"
                >
                  Reset
                </button>
              </div>
            </div>
          </div>
          <p
            class="text-lg ml-8"
          >
            Search for a patient
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <header
        class="bg-white shadow-md flex justify-between items-center"
      >
        <span
          class="cursor-pointer ml-3"
        >
          <a
            href="/"
          >
            <img
              src="/public/logo/ampath_Logo.png"
              width="180"
            />
          </a>
        </span>
        <nav
          class="md:mr-12"
        >
          <ul
            class="md:flex items-center gap-20"
          >
            <li
              class="text-sm"
            >
              Logged in as 
              <span
                class="uppercase"
              >
                <strong />
              </span>
            </li>
            <li
              class="text-lg flex gap-2 items-center hover:shadow-lg hover:bg-slate-300 p-2 rounded-sm cursor-pointer"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M17 7l-1.41 1.41L18.17 11H8v2h10.17l-2.58 2.58L17 17l5-5zM4 5h8V3H4c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h8v-2H4V5z"
                />
              </svg>
              Log out
            </li>
          </ul>
        </nav>
      </header>
    </div>
    <div
      class="bg-themeColor overflow-y-auto h-screen pt-10"
    >
      <div
        class="w-[80%] ml-[12%]"
      >
        <div
          class="w-[90%] mx-auto"
        >
          <div
            class="md:flex gap-10 m-4 mx-auto w-[95%] md:ml-20"
          >
            <input
              class="py-2 px-4 md:w-[60%] w-full outline-none rounded-xl shadow-lg"
              placeholder="Search patient by name"
              type="text"
              value=""
            />
            <div
              class="flex md:gap-11 gap-3 mt-4"
            >
              <button
                class="bg-blue-800/70 text-lg text-white py-2 px-12 rounded-xl hover:bg-white border hover:border-blue-800/70 hover:text-blue-800/70"
              >
                Search
              </button>
              <button
                class="bg-slate-50 text-lg text-red-600 border hover:border-red-500 hover:font-bold border-gray-300 py-2 px-12 rounded-xl"
              >
                Reset
              </button>
            </div>
          </div>
        </div>
        <p
          class="text-lg ml-8"
        >
          Search for a patient
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
